### ISSUE TITLE
Flavors: Add `multimarkdown` flavor for MultiMarkdown v7 specification

### ISSUE THREAD

**Issue by pygarap** [rvben/rumdl#399]
## Problem

rumdl flavors help avoid false positives for Markdown variants (MkDocs, MDX, Quarto, Obsidian). MultiMarkdown (v7 spec, largely compatible with v6) adds syntax that is not covered by the `standard` (CommonMark + GFM extensions) baseline, which currently causes unnecessary lint noise when linting MultiMarkdown documents.

## Proposal

Add a new flavor: `multimarkdown` (MultiMarkdown v7 specification), built on top of `standard`, with flavor-specific rule handling for MultiMarkdown-only constructs.

Configuration should work the same as existing flavors:

```toml
[global]
flavor = "multimarkdown"

[per-file-flavor]
"**/*.mmd" = "multimarkdown"
```

Auto-detect `.mmd` as `multimarkdown` when no flavor is configured.

## Supported patterns

At minimum, recognize these MultiMarkdown patterns so they do not trigger false positives:

### Metadata block at top of file

```text
---
Title: Example
Author: Jane Doe
...
```

(or without `---` / `...`, as long as it is a top-of-file `Key: Value` metadata block terminated by a blank line)

### Inline citations

```text
As per Doe.[#John Doe. *A Totally Fake Book 1*. Vanity Press, 2006.]
```

### Definition lists

```text
Apple
: Pomaceous fruit.

Orange
: The fruit of an evergreen tree.
```

### Table captions

Caption line immediately following a table:

```text
[Prototype table]
```

## Rule behavior changes

Implement flavor-specific handling (similar in spirit to `quarto`) so these constructs are not treated as broken links or other invalid Markdown:

| Rule | Standard behavior | `multimarkdown` behavior |
| --- | --- | --- |
| MD041 | First content line checks apply immediately | Skip MultiMarkdown metadata preamble when determining first heading |
| MD042 | May flag bracket-only constructs as empty links | Do not treat citations or table captions as links |
| MD051 | May attempt to validate fragments in citation-like syntax | Skip citation constructs |
| MD052 | May treat citations/captions as undefined reference links | Skip citation and caption constructs |

If other rules are impacted in practice, they can be added incrementally once we have concrete false-positive examples.

## Implementation checklist

- Add `multimarkdown` to the flavor enum and configuration validation (including `rumdl.schema.json`).
- Add `.mmd` to flavor auto-detection.
- Add docs:
  - Update `docs/flavors.md` quick reference table.
  - Add `docs/flavors/multimarkdown.md` (match the structure of existing flavor docs).
- Add tests/fixtures covering metadata, citations, definition lists, and table captions (including at least one `.mmd` file that previously produced false positives).

## References

- https://github.com/fletcher/MultiMarkdown-7/blob/develop/README.md
- https://fletcher.github.io/MultiMarkdown-6/MMD_Users_Guide.html
- https://fletcher.github.io/MultiMarkdown-7/api/index.html
- https://markedapp.com/help/MultiMarkdown_v5_Spec

**Comment by rvben** [comment]
Thanks for the detailed request, @pygarap. After investigating, we're going to hold off on MultiMarkdown support for now. The v7 spec is still pre-release, and the active user base is small relative to the flavors we currently support. Many MMD features (footnotes, definition lists) also overlap with Pandoc and existing flavor support.

In the meantime, `per-file-ignores` can suppress false positives for `.mmd` files:
```toml
[per-file-ignores]
"**/*.mmd" = ["MD041", "MD042", "MD051", "MD052"]
```

If MMD v7 reaches a stable release and adoption grows, we're happy to revisit.
